examples/hpc/dla-driver/src/tensor3.rs

```diff
@@ -0,0 +1,248 @@
+                _ => unimplemented!(),
+            };
+            out[i] = param;
+        }
+        out
+    }
+
+    /// Returns a reference to the element at the specified position
+    pub fn get(&self, channel: usize, row: usize, col: usize) -> Option<&T> {
+        self.data.get((channel, row, col))
+    }
+
+    /// Returns a mutable reference to the element at the specified position
+    pub fn get_mut(&mut self, channel: usize, row: usize, col: usize) -> Option<&mut T> {
+        self.data.get_mut((channel, row, col))
+    }
+
+    /// Sets the element at the specified position
+    pub fn set(
+        &mut self,
+        channel: usize,
+        row: usize,
+        col: usize,
+        value: T,
+    ) -> Result<(), &'static str> {
+        if let Some(elem) = self.data.get_mut((channel, row, col)) {
+            *elem = value;
+            Ok(())
+        } else {
+            Err("Index out of bounds")
+        }
+    }
+
+    /// Returns the dimensions of the array
+    pub fn dimensions(&self) -> (usize, usize, usize) {
+        (self.channels, self.height, self.width)
+    }
+
+    /// Gets the current order of the array
+    pub fn order(&self) -> Order3 {
+        self.order
+    }
+
+    /// Sets a new order for the array
+    pub fn transmute(&mut self, order: Order3) {
+        // Early return if already in order
+        if self.order == order {
+            return;
+        }
+
+        if self.order == Order3::CHW {
+            // Transmute to target order
+            let new_order: [usize; 3] = order.into();
+            self.data = self.data.clone().permuted_axes(new_order);
+            self.order = order;
+            return;
+        }
+
+        // Transmute to standard order
+        let std_order: [usize; 3] = self.order.into();
+        let std = self.data.clone().permuted_axes(std_order);
+
+        // Transmute to target order
+        let new_order: [usize; 3] = order.into();
+        self.data = std.permuted_axes(new_order);
+        self.order = order;
+    }
+
+    /// Converts the internal buffer to CHW order
+    fn to_chw_buffer(&self) -> Vec<T> {
+        self.to_buffer_with_order(Order3::CHW)
+    }
+
+    /// Converts the 3D array to a linear buffer according to the current order
+    pub fn to_buffer(&self) -> Vec<T> {
+        let mut buffer = Vec::with_capacity(self.channels * self.height * self.width);
+        for x in Array::from_iter(self.data.iter().cloned()) {
+            buffer.push(x)
+        }
+        buffer
+    }
+
+    /// Converts the 3D array to a linear buffer according to the specified order
+    pub fn to_buffer_with_order(&self, order: Order3) -> Vec<T> {
+        // If order is correct no need to permute
+        if order == self.order {
+            return self.to_buffer();
+        }
+
+        let mut data = self.clone();
+        data.transmute(order);
+        data.to_buffer()
+    }
+
+    /// Prints tensor in current order
+    pub fn print_tensor(&self) {
```

Comment:
This is fine for debug but on bare-metal, sprintln is typically very runtime dependent. I would generally avoid depending on sprintln directly from any driver components to enable easy porting, i.e., avoiding any `print_*` methods. This type of design is by the way exactly the reason why TVM runtime requires newlib; it's a design choice. Maybe more along the lines of _"something to keep in mind"_ than _"needs fix"_.

Fixable in many ways, all optional:

1. An easy alternative is to extract this method out of the struct.
2. Putting it behind a feature flag also makes this (arguably) easier to port.
3. One alternative would be to add `T: core::fmt::Write` as parameter then write into that instead of using sprintln. Then this could be used like so `tensor.write(&mut serial);`

Reply:
Yes, this is definitely a good guideline to follow. Removed all printing from tensors.